inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/mongodb/MongoDBTableSource.java

```diff
@@ -0,0 +1,341 @@
+                (RowType) physicalSchema.toPhysicalRowDataType().getLogicalType();
+        MetadataConverter[] metadataConverters = getMetadataConverters();
+        TypeInformation<RowData> typeInfo = scanContext.createTypeInformation(producedDataType);
+
+        DebeziumDeserializationSchema<RowData> deserializer =
+                new MongoDBConnectorDeserializationSchema(
+                        physicalDataType, metadataConverters, typeInfo, localTimeZone, metricOption);
+
+        String databaseList = null;
+        String collectionList = null;
+        if (StringUtils.isNotEmpty(database) && StringUtils.isNotEmpty(collection)) {
+            // explicitly specified database and collection.
+            if (!containsRegexMetaCharacters(database)
+                    && !containsRegexMetaCharacters(collection)) {
+                checkDatabaseNameValidity(database);
+                checkCollectionNameValidity(collection);
+                databaseList = database;
+                collectionList = database + "." + collection;
+            } else {
+                databaseList = database;
+                collectionList = collection;
+            }
+        } else if (StringUtils.isNotEmpty(database)) {
+            databaseList = database;
+        } else if (StringUtils.isNotEmpty(collection)) {
+            collectionList = collection;
+        } else {
+            // Watching all changes on the cluster by default, we do nothing here
+        }
+
+        if (enableParallelRead) {
+            MongoDBSourceBuilder<RowData> builder =
+                    MongoDBSource.<RowData>builder().hosts(hosts).deserializer(deserializer);
+
+            Optional.ofNullable(databaseList).ifPresent(builder::databaseList);
+            Optional.ofNullable(collectionList).ifPresent(builder::collectionList);
+            Optional.ofNullable(username).ifPresent(builder::username);
+            Optional.ofNullable(password).ifPresent(builder::password);
+            Optional.ofNullable(connectionOptions).ifPresent(builder::connectionOptions);
+            Optional.ofNullable(copyExisting).ifPresent(builder::copyExisting);
+            Optional.ofNullable(batchSize).ifPresent(builder::batchSize);
+            Optional.ofNullable(pollMaxBatchSize).ifPresent(builder::pollMaxBatchSize);
+            Optional.ofNullable(pollAwaitTimeMillis).ifPresent(builder::pollAwaitTimeMillis);
+            Optional.ofNullable(heartbeatIntervalMillis)
+                    .ifPresent(builder::heartbeatIntervalMillis);
+            Optional.ofNullable(splitMetaGroupSize).ifPresent(builder::splitMetaGroupSize);
+            Optional.ofNullable(splitSizeMB).ifPresent(builder::splitSizeMB);
+
+            return SourceProvider.of(builder.build());
+        } else {
+            com.ververica.cdc.connectors.mongodb.MongoDBSource.Builder<RowData> builder =
+                    com.ververica.cdc.connectors.mongodb.MongoDBSource.<RowData>builder()
```

Comment:
MongoDBSource is imported already

Reply:
The import class is com.ververica.cdc.connectors.**mongodb.source**.MongoDBSource ,

and here is  com.ververica.cdc.connectors.**mongodb**.MongoDBSource  ~